Gems/Atom/Tools/AtomToolsFramework/Code/Source/Viewport/ModularViewportCameraController.cpp

```diff
@@ -118,18 +118,20 @@ namespace AtomToolsFramework
     }
 
     // should the camera system respond to this particular event
-    static bool ShouldHandle(const AzFramework::ViewportControllerPriority priority, const bool exclusive)
+    static bool ShouldHandle(const AzFramework::ViewportControllerPriority priority, const bool exclusive, const bool handling)
     {
-        // ModernViewportCameraControllerInstance receives events at all priorities, it should only respond
-        // to normal priority events if it is not in 'exclusive' mode and when in 'exclusive' mode it should
-        // only respond to the highest priority events
+        // ModernViewportCameraControllerInstance receives events at all priorities, it should only respond to normal priority
+        // events if it is not in 'exclusive' mode and when in 'exclusive' mode it should only respond to the highest priority
+        // events, it should also respond to highest priority events while handling events (essentially when the camera system
+        // is 'active' and responding to inputs)
         return !exclusive && priority == AzFramework::ViewportControllerPriority::Normal ||
-            exclusive && priority == AzFramework::ViewportControllerPriority::Highest;
+            exclusive && priority == AzFramework::ViewportControllerPriority::Highest ||
+            handling && priority == AzFramework::ViewportControllerPriority::Highest;
```

Comment:
just verifying the logic here - should the first clause be "!(exclusive || handling)"?  Right now, if it's not exclusive, it looks like it will return true for Normal events whether or not it's handling events.  This doesn't match what the comments say.

Reply:
Hmm I think it's more likely the comment is wrong now... tried to update it but perhaps the first part is now incorrect. I think it should be...

> The camera controller should handle high priority events if it is either `exclusive` or `handling` otherwise if the camera is not exclusive, we only handle normal priority events...

I think in practice it likely doesn't really matter but it's possible your version is more correct, I'll try with that change and verify things still work as expected (if they don't think I think we actually do want that behavior and I'll update the comment 😛)

Reply:
We talked about this offline and have an idea to simplify the logic a bit to make it clearer what's going on by returning the priority, not a true/false value. Will see if I can update this and update the PR